utils/oscap-remediate-offline-service

```diff
@@ -0,0 +1,101 @@
+#!/bin/bash
+
+# SPDX-License-Identifier: GPL-2.0-or-later
+# Copyright (C) 2021 Red Hat Inc., Durham, North Carolina.
+#
+# Evgenii Kolesnikov <ekolesni@redhat.com>
+#
+# This is OpenSCAP's offline system remediation service wrapper script and
+# it is supposed to be executed by systemd and it is not sutable for manual
+# invocation.
+
+
+UCHAR_MAX=255
+SYSTEMD_SYSTEM_UPDATE_LINK="/system-update"
+OSCAP_REMEDIATE_CONF_BASENAME="oscap-remediate-offline.conf.sh"
+
+
+display() {
+	# Plymouth does not like messages longer than UCHAR_MAX
+	plymouth display-message --text="${1:0:${UCHAR_MAX}}"
+}
+
+log() {
+	echo "${1}" | logger -t oscap-remediate-offline -p ${2:-info}
+}
+
+die() {
+	log "${1}" error
+	exit 1
+}
+
+
+config=$(readlink -f "${SYSTEMD_SYSTEM_UPDATE_LINK}")
+
+if [[ "$(basename ${config})" != "${OSCAP_REMEDIATE_CONF_BASENAME}" ]]; then
+	log "The ${SYSTEMD_SYSTEM_UPDATE_LINK} symlink does not point to oscap offline remediation configuration, ignoring"
+	exit 0
+fi
+
+rm -f "${SYSTEMD_SYSTEM_UPDATE_LINK}"
+log "Removed ${SYSTEMD_SYSTEM_UPDATE_LINK}"
+log "Found config: ${config}"
+
+source "$config"
+log "OSCAP_REMEDIATE_DS: ${OSCAP_REMEDIATE_DS}"
+log "OSCAP_REMEDIATE_PROFILE_ID: ${OSCAP_REMEDIATE_PROFILE_ID}"
+log "OSCAP_REMEDIATE_LOGS_DIR: ${OSCAP_REMEDIATE_LOGS_DIR}"
+
+if [[ -f "${OSCAP_REMEDIATE_DS}" ]]; then
+	oscap xccdf validate --skip-schematron "${OSCAP_REMEDIATE_DS}"\
+		|| die "Invalid data stream file: ${OSCAP_REMEDIATE_DS}"
+
+	profile=${OSCAP_REMEDIATE_PROFILE_ID:-default}
+	profile_title_line=$(oscap info --profile "${profile}" "${OSCAP_REMEDIATE_DS}" | grep Title)\
+		|| die "Can not find profile: ${profile}"
+	profile_title=${profile_title_line#*Title: }
+else
+	die "The data stream file does not exists: ${OSCAP_REMEDIATE_DS}"
+fi
+log "Profile: ${profile} (${profile_title})"
+
+log_dir="${OSCAP_REMEDIATE_LOGS_DIR:-/var/lib/openscap/boot/$(date +%Y.%m.%d-%H:%M:%S)}"
+mkdir -p "${log_dir}" || die "Can not create the directory for logs: ${log_dir}"
+
+cd "${log_dir}"
+log "Logs and results directory: ${log_dir}"
+
+# Now we are good to go
+header="OpenSCAP is checking the system for compliance using"$'\n'"${profile_title}"$'\n\n'"Evaluating..."
+display "$header"
+
+command="oscap xccdf eval --profile ${OSCAP_REMEDIATE_PROFILE_ID}\
+	--results-arf arf-results.xml\
+	--report report.html\
+	--progress-full\
+	--remediate ${OSCAP_REMEDIATE_DS}"
+
+while IFS= read -r line; do
```

Comment:
Is there a need to manipulate `IFS`? `read` supports the `-d` option, would it make sense to use it to avoid interaction with IFS?

Reply:
Meh, yes and no. IFS has nothing to do with -d in this case, but is not needed here anymore. Nice catch.

Reply:
Fixed.